pkg/api/validation/validation.go

```diff
@@ -38,8 +38,7 @@ func validateVolumes(volumes []api.Volume) (util.StringSet, errs.ValidationError
 	allErrs := errs.ValidationErrorList{}
 
 	allNames := util.StringSet{}
-	for i := range volumes {
-		vol := &volumes[i] // so we can set default values
+	for i, vol := range volumes {
```

Comment:
Why pessimize when we don't really have to? I leave it to Yu-Ju - if it is a pain to convert back, skip it for now..